Error installing Matlab wraper

## Description

Following the github instructions I get the following error installing the matlab toolbox at the 100% mark after a make check which passed 100% of the test. 

Had an open issue which was similar that was closed but even though the other error message was resolved, and doesn't appear, I still get an error installing the wrapper.

## Steps to reproduce

1. Follow github instructions:
```
#!bash
mkdir build
cd build
cmake -DCMAKE_PREFIX_PATH=/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/boost_1_87_0 -DCMAKE_INSTALL_PREFIX=/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam_install -DGTSAM_INSTALL_MATLAB_TOOLBOX=ON ..
make check (optional, runs unit tests)
make install
```

make intstall - Error:
```
[100%] Built target TimeOfArrivalExample
[100%] Generating ../wrap/gtsam/gtsam_wrapper.cpp
[MatlabWrapper] Ignoring classes: ['gtsam::Point2', 'gtsam::Point3', 'gtsam::CustomFactor']
[100%] Building CXX object matlab/CMakeFiles/gtsam_matlab_wrapper.dir/__/wrap/gtsam/gtsam_wrapper.cpp.o
In file included from /usr/include/x86_64-linux-gnu/c++/9/bits/c++allocator.h:33,
                 from /usr/include/c++/9/bits/allocator.h:46,
                 from /usr/include/c++/9/string:41,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/base/types.h:28,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/global_includes.h:22,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/base/Vector.h:29,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/wrap/matlab.h:25,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/9/ext/new_allocator.h: In instantiation of ‘void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = gtsam::DiscreteConditional; _Args = {std::shared_ptr<gtsam::DiscreteConditional>}; _Tp = gtsam::DiscreteConditional]’:
/usr/include/c++/9/bits/alloc_traits.h:483:4:   required from ‘static void std::allocator_traits<std::allocator<_CharT> >::construct(std::allocator_traits<std::allocator<_CharT> >::allocator_type&, _Up*, _Args&& ...) [with _Up = gtsam::DiscreteConditional; _Args = {std::shared_ptr<gtsam::DiscreteConditional>}; _Tp = gtsam::DiscreteConditional; std::allocator_traits<std::allocator<_CharT> >::allocator_type = std::allocator<gtsam::DiscreteConditional>]’
/usr/include/c++/9/bits/shared_ptr_base.h:548:39:   required from ‘std::_Sp_counted_ptr_inplace<_Tp, _Alloc, _Lp>::_Sp_counted_ptr_inplace(_Alloc, _Args&& ...) [with _Args = {std::shared_ptr<gtsam::DiscreteConditional>}; _Tp = gtsam::DiscreteConditional; _Alloc = std::allocator<gtsam::DiscreteConditional>; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/9/bits/shared_ptr_base.h:679:16:   required from ‘std::__shared_count<_Lp>::__shared_count(_Tp*&, std::_Sp_alloc_shared_tag<_Alloc>, _Args&& ...) [with _Tp = gtsam::DiscreteConditional; _Alloc = std::allocator<gtsam::DiscreteConditional>; _Args = {std::shared_ptr<gtsam::DiscreteConditional>}; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/9/bits/shared_ptr_base.h:1344:71:   required from ‘std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::_Sp_alloc_shared_tag<_Tp>, _Args&& ...) [with _Alloc = std::allocator<gtsam::DiscreteConditional>; _Args = {std::shared_ptr<gtsam::DiscreteConditional>}; _Tp = gtsam::DiscreteConditional; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/9/bits/shared_ptr.h:359:59:   required from ‘std::shared_ptr<_Tp>::shared_ptr(std::_Sp_alloc_shared_tag<_Tp>, _Args&& ...) [with _Alloc = std::allocator<gtsam::DiscreteConditional>; _Args = {std::shared_ptr<gtsam::DiscreteConditional>}; _Tp = gtsam::DiscreteConditional]’
/usr/include/c++/9/bits/shared_ptr.h:701:14:   required from ‘std::shared_ptr<_Tp> std::allocate_shared(const _Alloc&, _Args&& ...) [with _Tp = gtsam::DiscreteConditional; _Alloc = std::allocator<gtsam::DiscreteConditional>; _Args = {std::shared_ptr<gtsam::DiscreteConditional>}]’
/usr/include/c++/9/bits/shared_ptr.h:717:39:   required from ‘std::shared_ptr<_Tp> std::make_shared(_Args&& ...) [with _Tp = gtsam::DiscreteConditional; _Args = {std::shared_ptr<gtsam::DiscreteConditional>}]’
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:14358:97:   required from here
/usr/include/c++/9/ext/new_allocator.h:146:4: error: no matching function for call to ‘gtsam::DiscreteConditional::DiscreteConditional(std::shared_ptr<gtsam::DiscreteConditional>)’
  146 |  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteBayesNet.h:21,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:20:
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:121:3: note: candidate: ‘gtsam::DiscreteConditional::DiscreteConditional(const gtsam::DecisionTreeFactor&, const gtsam::DecisionTreeFactor&, const gtsam::Ordering&)’
  121 |   DiscreteConditional(const DecisionTreeFactor& joint,
      |   ^~~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:121:3: note:   candidate expects 3 arguments, 1 provided
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:113:3: note: candidate: ‘gtsam::DiscreteConditional::DiscreteConditional(const gtsam::DecisionTreeFactor&, const gtsam::DecisionTreeFactor&)’
  113 |   DiscreteConditional(const DecisionTreeFactor& joint,
      |   ^~~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:113:3: note:   candidate expects 2 arguments, 1 provided
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:106:3: note: candidate: ‘gtsam::DiscreteConditional::DiscreteConditional(const DiscreteKey&, const string&)’
  106 |   DiscreteConditional(const DiscreteKey& key, const std::string& spec)
      |   ^~~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:106:3: note:   candidate expects 2 arguments, 1 provided
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:101:3: note: candidate: ‘gtsam::DiscreteConditional::DiscreteConditional(const DiscreteKey&, const gtsam::DiscreteKeys&, const string&)’
  101 |   DiscreteConditional(const DiscreteKey& key, const DiscreteKeys& parents,
      |   ^~~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:101:3: note:   candidate expects 3 arguments, 1 provided
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:87:3: note: candidate: ‘gtsam::DiscreteConditional::DiscreteConditional(const DiscreteKey&, const gtsam::DiscreteKeys&, const std::vector<double>&)’
   87 |   DiscreteConditional(const DiscreteKey& key, const DiscreteKeys& parents,
      |   ^~~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:87:3: note:   candidate expects 3 arguments, 1 provided
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:76:3: note: candidate: ‘gtsam::DiscreteConditional::DiscreteConditional(const DiscreteKey&, const gtsam::DiscreteKeys&, const Table&)’
   76 |   DiscreteConditional(const DiscreteKey& key, const DiscreteKeys& parents,
      |   ^~~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:76:3: note:   candidate expects 3 arguments, 1 provided
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:67:12: note: candidate: ‘gtsam::DiscreteConditional::DiscreteConditional(const gtsam::Signature&)’
   67 |   explicit DiscreteConditional(const Signature& signature);
      |            ^~~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:67:49: note:   no known conversion for argument 1 from ‘std::shared_ptr<gtsam::DiscreteConditional>’ to ‘const gtsam::Signature&’
   67 |   explicit DiscreteConditional(const Signature& signature);
      |                                ~~~~~~~~~~~~~~~~~^~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:63:3: note: candidate: ‘gtsam::DiscreteConditional::DiscreteConditional(size_t, const gtsam::DiscreteKeys&, const ADT&)’
   63 |   DiscreteConditional(size_t nFrontals, const DiscreteKeys& keys,
      |   ^~~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:63:3: note:   candidate expects 3 arguments, 1 provided
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:57:3: note: candidate: ‘gtsam::DiscreteConditional::DiscreteConditional(size_t, const gtsam::DiscreteFactor&)’
   57 |   DiscreteConditional(size_t nFrontals, const DiscreteFactor& f);
      |   ^~~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:57:3: note:   candidate expects 2 arguments, 1 provided
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:54:3: note: candidate: ‘gtsam::DiscreteConditional::DiscreteConditional()’
   54 |   DiscreteConditional() {}
      |   ^~~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:54:3: note:   candidate expects 0 arguments, 1 provided
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:37:20: note: candidate: ‘gtsam::DiscreteConditional::DiscreteConditional(const gtsam::DiscreteConditional&)’
   37 | class GTSAM_EXPORT DiscreteConditional
      |                    ^~~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:37:20: note:   no known conversion for argument 1 from ‘std::shared_ptr<gtsam::DiscreteConditional>’ to ‘const gtsam::DiscreteConditional&’
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:37:20: note: candidate: ‘gtsam::DiscreteConditional::DiscreteConditional(gtsam::DiscreteConditional&&)’
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DiscreteConditional.h:37:20: note:   no known conversion for argument 1 from ‘std::shared_ptr<gtsam::DiscreteConditional>’ to ‘gtsam::DiscreteConditional&&’
In file included from /usr/include/x86_64-linux-gnu/c++/9/bits/c++allocator.h:33,
                 from /usr/include/c++/9/bits/allocator.h:46,
                 from /usr/include/c++/9/string:41,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/base/types.h:28,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/global_includes.h:22,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/base/Vector.h:29,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/wrap/matlab.h:25,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/9/ext/new_allocator.h: In instantiation of ‘void __gnu_cxx::new_allocator<_Tp>::construct(_Up*, _Args&& ...) [with _Up = gtsam::DecisionTreeFactor; _Args = {std::shared_ptr<gtsam::DiscreteFactor>}; _Tp = gtsam::DecisionTreeFactor]’:
/usr/include/c++/9/bits/alloc_traits.h:483:4:   required from ‘static void std::allocator_traits<std::allocator<_CharT> >::construct(std::allocator_traits<std::allocator<_CharT> >::allocator_type&, _Up*, _Args&& ...) [with _Up = gtsam::DecisionTreeFactor; _Args = {std::shared_ptr<gtsam::DiscreteFactor>}; _Tp = gtsam::DecisionTreeFactor; std::allocator_traits<std::allocator<_CharT> >::allocator_type = std::allocator<gtsam::DecisionTreeFactor>]’
/usr/include/c++/9/bits/shared_ptr_base.h:548:39:   required from ‘std::_Sp_counted_ptr_inplace<_Tp, _Alloc, _Lp>::_Sp_counted_ptr_inplace(_Alloc, _Args&& ...) [with _Args = {std::shared_ptr<gtsam::DiscreteFactor>}; _Tp = gtsam::DecisionTreeFactor; _Alloc = std::allocator<gtsam::DecisionTreeFactor>; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/9/bits/shared_ptr_base.h:679:16:   required from ‘std::__shared_count<_Lp>::__shared_count(_Tp*&, std::_Sp_alloc_shared_tag<_Alloc>, _Args&& ...) [with _Tp = gtsam::DecisionTreeFactor; _Alloc = std::allocator<gtsam::DecisionTreeFactor>; _Args = {std::shared_ptr<gtsam::DiscreteFactor>}; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/9/bits/shared_ptr_base.h:1344:71:   required from ‘std::__shared_ptr<_Tp, _Lp>::__shared_ptr(std::_Sp_alloc_shared_tag<_Tp>, _Args&& ...) [with _Alloc = std::allocator<gtsam::DecisionTreeFactor>; _Args = {std::shared_ptr<gtsam::DiscreteFactor>}; _Tp = gtsam::DecisionTreeFactor; __gnu_cxx::_Lock_policy _Lp = __gnu_cxx::_S_atomic]’
/usr/include/c++/9/bits/shared_ptr.h:359:59:   required from ‘std::shared_ptr<_Tp>::shared_ptr(std::_Sp_alloc_shared_tag<_Tp>, _Args&& ...) [with _Alloc = std::allocator<gtsam::DecisionTreeFactor>; _Args = {std::shared_ptr<gtsam::DiscreteFactor>}; _Tp = gtsam::DecisionTreeFactor]’
/usr/include/c++/9/bits/shared_ptr.h:701:14:   required from ‘std::shared_ptr<_Tp> std::allocate_shared(const _Alloc&, _Args&& ...) [with _Tp = gtsam::DecisionTreeFactor; _Alloc = std::allocator<gtsam::DecisionTreeFactor>; _Args = {std::shared_ptr<gtsam::DiscreteFactor>}]’
/usr/include/c++/9/bits/shared_ptr.h:717:39:   required from ‘std::shared_ptr<_Tp> std::make_shared(_Args&& ...) [with _Tp = gtsam::DecisionTreeFactor; _Args = {std::shared_ptr<gtsam::DiscreteFactor>}]’
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:14999:86:   required from here
/usr/include/c++/9/ext/new_allocator.h:146:4: error: no matching function for call to ‘gtsam::DecisionTreeFactor::DecisionTreeFactor(std::shared_ptr<gtsam::DiscreteFactor>)’
  146 |  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:19:
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:120:14: note: candidate: ‘gtsam::DecisionTreeFactor::DecisionTreeFactor(const gtsam::DiscreteConditional&)’
  120 |     explicit DecisionTreeFactor(const DiscreteConditional& c);
      |              ^~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:120:60: note:   no known conversion for argument 1 from ‘std::shared_ptr<gtsam::DiscreteFactor>’ to ‘const gtsam::DiscreteConditional&’
  120 |     explicit DecisionTreeFactor(const DiscreteConditional& c);
      |                                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~^
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:116:5: note: candidate: ‘gtsam::DecisionTreeFactor::DecisionTreeFactor(const DiscreteKey&, const std::vector<double>&)’
  116 |     DecisionTreeFactor(const DiscreteKey& key, const std::vector<double>& row)
      |     ^~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:116:5: note:   candidate expects 2 arguments, 1 provided
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:112:5: note: candidate: ‘template<class SOURCE> gtsam::DecisionTreeFactor::DecisionTreeFactor(const DiscreteKey&, SOURCE)’
  112 |     DecisionTreeFactor(const DiscreteKey& key, SOURCE table)
      |     ^~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:112:5: note:   template argument deduction/substitution failed:
In file included from /usr/include/x86_64-linux-gnu/c++/9/bits/c++allocator.h:33,
                 from /usr/include/c++/9/bits/allocator.h:46,
                 from /usr/include/c++/9/string:41,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/base/types.h:28,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/global_includes.h:22,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/base/Vector.h:29,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/wrap/matlab.h:25,
                 from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:1:
/usr/include/c++/9/ext/new_allocator.h:146:4: note:   candidate expects 2 arguments, 1 provided
  146 |  { ::new((void *)__p) _Up(std::forward<_Args>(__args)...); }
      |    ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/build/wrap/gtsam/gtsam_wrapper.cpp:19:
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:108:5: note: candidate: ‘gtsam::DecisionTreeFactor::DecisionTreeFactor(const gtsam::DiscreteKeys&, const string&)’
  108 |     DecisionTreeFactor(const DiscreteKeys& keys, const std::string& table);
      |     ^~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:108:5: note:   candidate expects 2 arguments, 1 provided
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:87:5: note: candidate: ‘gtsam::DecisionTreeFactor::DecisionTreeFactor(const gtsam::DiscreteKeys&, const std::vector<double>&)’
   87 |     DecisionTreeFactor(const DiscreteKeys& keys,
      |     ^~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:87:5: note:   candidate expects 2 arguments, 1 provided
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:66:5: note: candidate: ‘gtsam::DecisionTreeFactor::DecisionTreeFactor(const gtsam::DiscreteKeys&, const ADT&)’
   66 |     DecisionTreeFactor(const DiscreteKeys& keys, const ADT& potentials);
      |     ^~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:66:5: note:   candidate expects 2 arguments, 1 provided
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:63:5: note: candidate: ‘gtsam::DecisionTreeFactor::DecisionTreeFactor()’
   63 |     DecisionTreeFactor();
      |     ^~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:63:5: note:   candidate expects 0 arguments, 1 provided
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:45:22: note: candidate: ‘gtsam::DecisionTreeFactor::DecisionTreeFactor(const gtsam::DecisionTreeFactor&)’
   45 |   class GTSAM_EXPORT DecisionTreeFactor : public DiscreteFactor,
      |                      ^~~~~~~~~~~~~~~~~~
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:45:22: note:   no known conversion for argument 1 from ‘std::shared_ptr<gtsam::DiscreteFactor>’ to ‘const gtsam::DecisionTreeFactor&’
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:45:22: note: candidate: ‘gtsam::DecisionTreeFactor::DecisionTreeFactor(gtsam::DecisionTreeFactor&&)’
/Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam/discrete/DecisionTreeFactor.h:45:22: note:   no known conversion for argument 1 from ‘std::shared_ptr<gtsam::DiscreteFactor>’ to ‘gtsam::DecisionTreeFactor&&’
make[2]: *** [matlab/CMakeFiles/gtsam_matlab_wrapper.dir/build.make:96: matlab/CMakeFiles/gtsam_matlab_wrapper.dir/__/wrap/gtsam/gtsam_wrapper.cpp.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:34652: matlab/CMakeFiles/gtsam_matlab_wrapper.dir/all] Error 2
make: *** [Makefile:166: all] Error 2

```

## Expected behavior

Install Matlab Wrapper

## Environment

Using a conda environment
gcc (Ubuntu 9.4.0-1ubuntu1~20.04.2) 9.4.0
Python 3.12.4
R2024b Update 1 (24.2.0.2740171) 64-bit (glnxa64)
cmake version 3.30.5
boost_1.87.0

## Additional information

Build results:
```
-- The CXX compiler identification is GNU 9.4.0
-- The C compiler identification is GNU 9.4.0
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- GTSAM_POSE3_EXPMAP=ON, enabling GTSAM_ROT3_EXPMAP as well
-- Performing Test COMPILER_HAS_WSUGGEST_OVERRIDE
-- Performing Test COMPILER_HAS_WSUGGEST_OVERRIDE - Success
-- Performing Test COMPILER_HAS_WMISSING_OVERRIDE
-- Performing Test COMPILER_HAS_WMISSING_OVERRIDE - Failed
-- Found Boost: /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/boost_1_87_0/boost_install_folder/lib/cmake/Boost-1.87.0/BoostConfig.cmake (found suitable version "1.87.0", minimum required is "1.65") found components: serialization system filesystem thread program_options date_time timer chrono regex
-- Found Eigen version: 3.4.0
CMake Deprecation Warning at gtsam/3rdparty/metis/CMakeLists.txt:1 (cmake_minimum_required):
  Compatibility with CMake < 3.5 will be removed from a future version of
  CMake.

  Update the VERSION argument <min> value or use a ...<max> suffix to tell
  CMake that the project does not need compatibility with older versions.


-- Looking for execinfo.h
-- Looking for execinfo.h - found
-- Looking for getline
-- Looking for getline - found
-- checking for thread-local storage - found
-- Could NOT find MKL (missing: MKL_INCLUDE_DIR MKL_LIBRARIES)
-- Found OpenMP_C: -fopenmp (found version "4.5")
-- Found OpenMP_CXX: -fopenmp (found version "4.5")
-- Found OpenMP: TRUE (found version "4.5")
-- Found Python3: /Data/miguelcruz/miniconda3/envs/gtsam/bin/python3.12 (found version "3.12.4") found components: Interpreter Development Development.Module Development.Embed
-- Building 3rdparty
-- Could NOT find GeographicLib (missing: GeographicLib_LIBRARY_DIRS GeographicLib_LIBRARIES GeographicLib_INCLUDE_DIRS)
-- Building base
-- Building basis
-- Building constrained
-- Building geometry
-- Building inference
-- Building symbolic
-- Building discrete
-- Building hybrid
-- Building linear
-- Building nonlinear
-- Building sam
-- Building sfm
-- Building slam
-- Building navigation
-- GTSAM Version: 4.3a0
-- Install prefix: /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam_install
-- Building GTSAM - as a SHARED library
-- Building base_unstable
-- Building geometry_unstable
-- Building linear_unstable
-- Building discrete_unstable
-- Building dynamics_unstable
-- Building nonlinear_unstable
-- Building slam_unstable
-- Building partition_unstable
-- GTSAM_UNSTABLE Version: 4.3a0
-- Install prefix: /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam_install
-- Found Python: /Data/miguelcruz/miniconda3/envs/gtsam/bin/python3.12 (found suitable exact version "3.12.4") found components: Interpreter Development Development.Module Development.Embed
-- gtwrap Package config : /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam_install/lib/cmake/gtwrap
-- gtwrap version        : 1.0
-- gtwrap CMake path     : /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam_install/lib/cmake/gtwrap
-- gtwrap library path   : /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam_install/lib/gtwrap
-- gtwrap binary path    : /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam_install/bin/gtwrap
-- gtwrap header path    : /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam_install/include/gtwrap
-- Found Matlab: /Data/miguelcruz/matlab (found version "24.2.0.2740171") found components: MEX_COMPILER
-- Building wrap module gtsam
CMake Warning at wrap/cmake/MatlabWrap.cmake:225 (find_package):
  By not providing "FindPythonInterp.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "PythonInterp", but CMake did not find one.

  Could not find a package configuration file provided by "PythonInterp"
  (requested version 3.12.4) with any of the following names:

    PythonInterpConfig.cmake
    pythoninterp-config.cmake

  Add the installation prefix of "PythonInterp" to CMAKE_PREFIX_PATH or set
  "PythonInterp_DIR" to a directory containing one of the above files.  If
  "PythonInterp" provides a separate development package or SDK, be sure it
  has been installed.
Call Stack (most recent call first):
  wrap/cmake/MatlabWrap.cmake:92 (wrap_library_internal)
  wrap/cmake/MatlabWrap.cmake:68 (wrap_and_install_library)
  matlab/CMakeLists.txt:89 (matlab_wrap)


CMake Warning at wrap/cmake/MatlabWrap.cmake:226 (find_package):
  By not providing "FindPythonLibs.cmake" in CMAKE_MODULE_PATH this project
  has asked CMake to find a package configuration file provided by
  "PythonLibs", but CMake did not find one.

  Could not find a package configuration file provided by "PythonLibs"
  (requested version 3.12.4) with any of the following names:

    PythonLibsConfig.cmake
    pythonlibs-config.cmake

  Add the installation prefix of "PythonLibs" to CMAKE_PREFIX_PATH or set
  "PythonLibs_DIR" to a directory containing one of the above files.  If
  "PythonLibs" provides a separate development package or SDK, be sure it has
  been installed.
Call Stack (most recent call first):
  wrap/cmake/MatlabWrap.cmake:92 (wrap_library_internal)
  wrap/cmake/MatlabWrap.cmake:68 (wrap_and_install_library)
  matlab/CMakeLists.txt:89 (matlab_wrap)


-- Installing Matlab Toolbox to /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/gtsam_install/gtsam_toolbox
-- Wrote /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/build/GTSAMConfig.cmake
-- Wrote /Data/miguelcruz/SingleImageDepthEstimation/prepareDS_registration/gtsam/gtsam/build/GTSAM_UNSTABLEConfig.cmake
-- Found Doxygen: /usr/bin/doxygen (found version "1.8.17") found components: doxygen missing components: dot
-- ===============================================================
-- ================  Configuration Options  ======================
--  CMAKE_CXX_COMPILER_ID type                       : GNU
--  CMAKE_CXX_COMPILER_VERSION                       : 9.4.0
--  CMake version                                    : 3.30.5
--  CMake generator                                  : Unix Makefiles
--  CMake build tool                                 : /usr/bin/make
-- Build flags
--  Build Tests                                      : Enabled
--  Build examples with 'make all'                   : Enabled
--  Build timing scripts with 'make all'             : Disabled
--  Build Docs                                       : Enabled
--  Build shared GTSAM libraries                     : Enabled
--  Put build type in library name                   : Enabled
--  Build libgtsam_unstable                          : Enabled
--  Build GTSAM unstable Python                      : Enabled
--  Build MATLAB Toolbox for unstable                : Disabled
--  Build for native architecture                    : Disabled
--  Build type                                       : Release
--  C compilation flags                              :  -O3 -DNDEBUG
--  C++ compilation flags                            :  -O3 -DNDEBUG
--  Enable Boost serialization                       : ON
--  GTSAM_COMPILE_FEATURES_PUBLIC                    : cxx_std_17
--  GTSAM_COMPILE_OPTIONS_PUBLIC                     :
--  GTSAM_COMPILE_DEFINITIONS_PUBLIC                 :
--  GTSAM_COMPILE_OPTIONS_PUBLIC_RELEASE             :
--  GTSAM_COMPILE_DEFINITIONS_PUBLIC_RELEASE         :
--  Use System Eigen                                 : OFF (Using version: 3.4.0)
--  Use System Metis                                 : OFF
--  Using Boost version                              : 1.87.0
--  Use Intel TBB                                    : Yes (Version: 2020.1)
--  Eigen will use MKL                               : MKL not found
--  Eigen will use MKL and OpenMP                    : OpenMP found but GTSAM_WITH_EIGEN_MKL is disabled
--  Default allocator                                : TBB
--  Cheirality exceptions enabled                    : YES
--  Build with ccache                                : No
-- Packaging flags
--  CPack Source Generator                           : TGZ
--  CPack Generator                                  : TGZ
-- GTSAM flags
--  Quaternions as default Rot3                      : Disabled
--  Runtime consistency checking                     : Disabled
--  Build with Memory Sanitizer                      : Disabled
--  Rot3 retract is full ExpMap                      : Enabled
--  Pose3 retract is full ExpMap                     : Enabled
--  Enable branch merging in DecisionTree            : Enabled
--  Enable timing machinery                          : Disabled
--  Allow features deprecated in GTSAM 4.3           : Enabled
--  Metis-based Nested Dissection                    : Enabled
--  Use tangent-space preintegration                 : Enabled
-- MATLAB toolbox flags
--  Install MATLAB toolbox                           : Enabled
--  MATLAB root                                      : /Data/miguelcruz/matlab
--  MEX binary                                       : /Data/miguelcruz/matlab/bin/glnxa64/mex
-- Python toolbox flags
--  Build Python module with pybind                  : Disabled
-- ===============================================================
-- Configuring done (3.2s)
-- Generating done (1.4s)
```

make check results:
```
(...)
100% tests passed, 0 tests failed out of 301

Total Test time (real) =   9.28 sec
[100%] Built target check
```

--- a/gtsam/discrete/discrete.i
+++ b/gtsam/discrete/discrete.i
@@ -288,7 +288,7 @@ class DiscreteBayesTree {
   const DiscreteBayesTreeClique* clique(size_t j) const;
   size_t numCachedSeparatorMarginals() const;
 
-  gtsam::DiscreteConditional marginalFactor(size_t key) const;
+  gtsam::DiscreteConditional* marginalFactor(size_t key) const;
   gtsam::DiscreteFactorGraph* joint(size_t j1, size_t j2) const;
   gtsam::DiscreteBayesNet* jointBayesNet(size_t j1, size_t j2) const;
 

--- a/gtsam/discrete/discrete.i
+++ b/gtsam/discrete/discrete.i
@@ -369,7 +369,7 @@ class DiscreteFactorGraph {
   void print(string s = "") const;
   bool equals(const gtsam::DiscreteFactorGraph& fg, double tol = 1e-9) const;
 
-  gtsam::DecisionTreeFactor product() const;
+  gtsam::DiscreteFactor* product() const;
   double operator()(const gtsam::DiscreteValues& values) const;
   gtsam::DiscreteValues optimize() const;
 

